readium/streamer/src/main/java/org/readium/r2/streamer/parser/PublicationParser.kt

```diff
@@ -6,15 +6,26 @@
 
 package org.readium.r2.streamer.parser
 
+import org.readium.r2.shared.error.SimpleError
+import org.readium.r2.shared.error.ThrowableError
+import org.readium.r2.shared.error.Try
+import org.readium.r2.shared.fetcher.Fetcher
 import org.readium.r2.shared.publication.Publication
-import org.readium.r2.shared.publication.asset.PublicationAsset
+import org.readium.r2.shared.resource.Resource
 import org.readium.r2.shared.util.logging.WarningLogger
+import org.readium.r2.shared.util.mediatype.MediaType
 
 /**
  *  Parses a Publication from an asset.
  */
 interface PublicationParser {
 
+    data class Asset(
+        val name: String,
+        val mediaType: MediaType,
+        val fetcher: Fetcher
+    )
```

Comment:
This looks very similar to `ContentProtection.Asset`. Is it something that we should refactor into a `PublicationAsset` (with the `onCreatePublication` closure)?